### PR TITLE
3.0.3 release

### PR DESCRIPTION
**Pull request by kennethmhc**
This PR adds/fixes/changes...
- please summarize your changes to the code 
- and make sure to include all changes to user-facing APIs

JIRA Issue: -

Priority for Review: -

Related PRs: -

**How Has This Been Tested?**

- [ ] Unit Tests
- [ ] Integration Tests
- [ ] Manual Tests on VM

**Checklist For The Assigned Reviewer:**

```
- [ ] Checked if merge conflicts with master exist
- [ ] Checked if stylechecks for Java and Python pass
- [ ] Checked if all docstrings were added and/or updated appropriately
- [ ] Ran spellcheck on docstring
- [ ] Checked if guides & concepts need to be updated
- [ ] Checked if naming conventions for parameters and variables were followed
- [ ] Checked if private methods are properly declared and used
- [ ] Checked if hard-to-understand areas of code are commented
- [ ] Checked if tests are effective
- [ ] Built and deployed changes on dev VM and tested manually
- [x] (Checked if all type annotations were added and/or updated appropriately)
```

### CODE REVIEW

**Review by jimdowling — APPROVED** [review]

File: python/hsfs/constructor/query.py [hunk 1]
```diff
@@ -169,23 +169,29 @@ def as_of(self, wallclock_time):
...
+
         for join in self._joins:
             join.query.left_feature_group_end_time = wallclock_timestamp
         self.left_feature_group_end_time = wallclock_timestamp
         return self
 
     def pull_changes(self, wallclock_start_time, wallclock_end_time):
-        self.left_feature_group_start_time = util.get_timestamp_from_date_string(
+        """
+        !!! warning "Deprecated"
```
Comment (tdoehmen) [inline python/hsfs/constructor/query.py:187]: This deprecation warning slipped in. Wa have to remove it

**Comment by tdoehmen** [comment]
Maybe we should add https://github.com/logicalclocks/feature-store-api/pull/791 as well

**Comment by tdoehmen** [comment]
It is essential that all changes are tested for backwards compatability with:
- 3.0.0 backend
- app with hsfs-3.0.2 docker base image

File: python/hsfs/constructor/query.py (hunk 1 above)
Comment (tdoehmen) [reply python/hsfs/constructor/query.py:187]: made a PR https://github.com/kennethmhc/feature-store-api/pull/2

**Comment by tdoehmen** [comment]
> Maybe we should add #791 as well

https://github.com/kennethmhc/feature-store-api/pull/2

**Comment by robzor92** [comment]
https://github.com/logicalclocks/feature-store-api/commit/ca0434126733053a29428dd826da72064e3ae42c should be added to this release also.

File: python/hsfs/feature_group.py [hunk 2]
```diff
@@ -1263,16 +1273,19 @@ def validate(
...
         feature store.
 
         Statistics are only computed for data in the offline storage of the feature
         group.
 
         # Arguments
-            wallclock_time: Date string in the format of "YYYYMMDD" or "YYYYMMDDhhmmss".
-                Only valid if feature group is time travel enabled. If specified will recompute statistics on
+            wallclock_time: datatime.datetime, datetime.date, unix timestamp in seconds (int), or string. The String should
+                be formatted in one of the following formats `%Y%m%d`, `%Y%m%d%H`, `%Y%m%d%H%M`, `%Y%m%d%H%M%S`,
```
Comment (jimdowling) [inline python/hsfs/feature_group.py:1287]: Why do we support these unreadable timestamp formats - not good for debugging. 
`%Y%m%d%H%M`
Can we also add common ones like ISO 86001
https://en.wikipedia.org/wiki/ISO_8601

`%Y-%m-%d %H:%M`

Comment (jimdowling) [reply python/hsfs/feature_group.py:1287]: ISO 8601 tackles this uncertainty by setting out an internationally agreed way to represent dates:

YYYY-MM-DD
https://www.iso.org/iso-8601-date-and-time-format.html

File: python/hsfs/feature_view.py [hunk 3]
```diff
@@ -149,13 +171,19 @@ def init_batch_scoring(
...
+        end_time: Optional[Union[str, int, datetime, date]] = None,
     ):
         """Get a query string of batch query.
 
         # Arguments
-            start_time: Optional. Start time of the batch query.
-            end_time: Optional. End time of the batch query.
+            start_time: Optional. Start time of the batch query. datatime.datetime, datetime.date, unix timestamp in seconds (int), or string.
+                The String should be formatted in one of the following formats `%Y%m%d`, `%Y%m%d%H`, `%Y%m%d%H%M`,
+                `%Y%m%d%H%M%S`, or `%Y%m%d%H%M%S%f`.
```
Comment (jimdowling) [inline python/hsfs/feature_view.py:183]: iso-8601

File: python/hsfs/feature_view.py [hunk 4]
```diff
@@ -256,11 +284,14 @@ def preview_feature_vectors(self, n: int, external: Optional[bool] = None):
...
         """
-        start_time: timestamp in second or wallclock_time: Datetime string. The String should be formatted in one of the
-            following formats `%Y%m%d`, `%Y%m%d%H`, `%Y%m%d%H%M`, `%Y%m%d%H%M%S`, or `%Y%m%d%H%M%S%f`.
-        end_time: timestamp in second or wallclock_time: Datetime string. The String should be formatted in one of the
-            following formats `%Y%m%d`, `%Y%m%d%H`, `%Y%m%d%H%M`, `%Y%m%d%H%M%S`,  or `%Y%m%d%H%M%S%f`.
-        read_options: User provided read options. Defaults to `{}`.
+        # Arguments
+            start_time: datatime.datetime, datetime.date, unix timestamp in seconds (int), or string. The String should be
+                formatted in one of the following formats `%Y%m%d`, `%Y%m%d%H`, `%Y%m%d%H%M`, `%Y%m%d%H%M%S`,
+                or `%Y%m%d%H%M%S%f`.
```
Comment (jimdowling) [inline python/hsfs/feature_view.py:290]: iso-8601

File: python/hsfs/feature_view.py [hunk 5]
```diff
@@ -421,14 +454,18 @@ def create_train_test_split(
...
             test_size: size of test set.
-            train_start: timestamp in second or wallclock_time: Datetime string. The String should be formatted in one of the
-                following formats `%Y%m%d`, `%Y%m%d%H`, `%Y%m%d%H%M`, `%Y%m%d%H%M%S`, or `%Y%m%d%H%M%S%f`.
-            train_end: timestamp in second or wallclock_time: Datetime string. The String should be formatted in one of the
-                following formats `%Y%m%d`, `%Y%m%d%H`, `%Y%m%d%H%M`, `%Y%m%d%H%M%S`,  or `%Y%m%d%H%M%S%f`.
-            test_start: timestamp in second or wallclock_time: Datetime string. The String should be formatted in one of the
-                following formats `%Y%m%d`, `%Y%m%d%H`, `%Y%m%d%H%M`, `%Y%m%d%H%M%S`, or `%Y%m%d%H%M%S%f`.
-            test_end: timestamp in second or wallclock_time: Datetime string. The String should be formatted in one of the
-                following formats `%Y%m%d`, `%Y%m%d%H`, `%Y%m%d%H%M`, `%Y%m%d%H%M%S`,  or `%Y%m%d%H%M%S%f`.
+            train_start: datatime.datetime, datetime.date, unix timestamp in seconds (int), or string. The String should
```
Comment (jimdowling) [inline python/hsfs/feature_view.py:457]: iso-8601

File: python/hsfs/feature_view.py [hunk 6]
```diff
@@ -746,14 +793,18 @@ def train_test_split(
...
             test_size: size of test set. Should be between 0 and 1.
-            train_start: timestamp in second or wallclock_time: Datetime string. The String should be formatted in one of the
-                following formats `%Y%m%d`, `%Y%m%d%H`, `%Y%m%d%H%M`, `%Y%m%d%H%M%S`, or `%Y%m%d%H%M%S%f`.
-            train_end: timestamp in second or wallclock_time: Datetime string. The String should be formatted in one of the
-                following formats `%Y%m%d`, `%Y%m%d%H`, `%Y%m%d%H%M`, `%Y%m%d%H%M%S`,  or `%Y%m%d%H%M%S%f`.
-            test_start: timestamp in second or wallclock_time: Datetime string. The String should be formatted in one of the
-                following formats `%Y%m%d`, `%Y%m%d%H`, `%Y%m%d%H%M`, `%Y%m%d%H%M%S`, or `%Y%m%d%H%M%S%f`.
-            test_end: timestamp in second or wallclock_time: Datetime string. The String should be formatted in one of the
-                following formats `%Y%m%d`, `%Y%m%d%H`, `%Y%m%d%H%M`, `%Y%m%d%H%M%S`,  or `%Y%m%d%H%M%S%f`.
+            train_start: datatime.datetime, datetime.date, unix timestamp in seconds (int), or string. The String should
```
Comment (jimdowling) [inline python/hsfs/feature_view.py:796]: iso-8601

**Review by jimdowling — APPROVED** [review]
A

File: python/hsfs/feature_group.py (hunk 2 above)
Comment (tdoehmen) [reply python/hsfs/feature_group.py:1287]: This format is supported. We'd just need to update the docstrings, as discussed here https://github.com/logicalclocks/feature-store-api/pull/783#discussion_r976294576 There is also an open Jira, as part of which this schould be taken care of.

**Pull request merged**